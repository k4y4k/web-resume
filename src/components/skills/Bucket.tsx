import React from "react";

interface BucketTypes {
  skills: string[];
}

const Bucket = ({ skills }: BucketTypes): JSX.Element | null => {
  if (skills.length === 1 && skills[0] === "") return null;

  return (
    <div data-testid="skillsBucket">
      {skills.map((el) => (
        <p
          style={{
            marginRight: "0.66rem",
            marginBottom: "0.2rem",
            display: "inline-block",
            lineHeight: "1.15",
          }}
          key={el}
        >
          {`${el}`}
        </p>
      ))}
    </div>
  );
};

export default Bucket;
