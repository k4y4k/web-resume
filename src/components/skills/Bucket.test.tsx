import * as React from 'react'
import { render, screen } from '@testing-library/react'
import Bucket from './Bucket'

describe('<Bucket />', () => {
  test('displays nothing for no data', () => {
    render(<Bucket skills={[]} />)

    const skills = screen.getByTestId('skillsBucket')

    expect(skills).toHaveTextContent('')
  })

  test('alphabetises input', () => {
    render(
      <Bucket
        skills={['JavaScript', 'React', 'A Minimal Caffeine Dependency']}
      />
    )

    const skills = screen.getByTestId('skillsBucket')
    expect(skills).toHaveTextContent(
      'A Minimal Caffeine Dependency JavaScript React'
    )
  })
})
