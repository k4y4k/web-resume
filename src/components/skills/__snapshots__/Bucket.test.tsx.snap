// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bucket /> alphabetises input 1`] = `
<div
  data-testid="skillsBucket"
>
  <span>
    A Minimal Caffeine Dependency 
  </span>
  <span>
    JavaScript 
  </span>
  <span>
    React 
  </span>
</div>
`;

exports[`<Bucket /> displays nothing for no data 1`] = `null`;
